### PR TITLE
executor: add placeholder count check in prepare stage

### PR DESCRIPTION
**Pull request by lysu**
## What have you changed? (mandatory)

Now, client will meet error when input parameter placeholder count > 65535 in exec stage
this PR add placeholder num check in prepare stage to return error more quick and more evident and make it compatible with mysql.

## What is the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

- uint tests
-  manual tests

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

no

## Does this PR affect tidb-ansible update? (mandatory)

no

## Does this PR need to be added to the release notes? (mandatory)

no

## Add a few positive/negative examples (optional)

run

```
package main

import "database/sql"

import (
	"bytes"
	"fmt"
	_ "github.com/go-sql-driver/mysql"
)

func main() {
	db, err := sql.Open("mysql", "root:root@tcp(127.0.0.1:3306)/test")
	if err != nil {
		panic(err)
	}
	defer db.Close()

	_, err = db.Exec("drop table if exists t")
	if err != nil {
		panic(err)
	}

	_, err = db.Exec("create table t (v int)")
	if err != nil {
		panic(err)
	}

	const pc = 65535 + 2

	var param [pc]interface{}
	param[0] = 0
	var sqlBuilder bytes.Buffer
	sqlBuilder.WriteString("insert into t values (?)")
	for i := 0; i < pc-1; i++ {
		sqlBuilder.WriteString(",(?)")
		param[i+1] = i
	}

	stmt, err := db.Prepare(sqlBuilder.String())
	if err != nil {
		panic(err)
	}

	result, err := stmt.Exec(param[:]...)
	if err != nil {
		panic(err)
	}

	fmt.Println(result)

}
```

expect:

```
panic: Error 1390: Prepared statement contains too many placeholders

goroutine 1 [running]:
main.main()
	test/main.go:41 +0x3d2
```

but got:

```
panic: sql: expected 1 arguments, got 65537

goroutine 1 [running]:
main.main()
	test/main.go:46 +0x3c2
```

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/7162)

### CODE REVIEW

File: ddl/table.go [hunk 1]
```diff
@@ -203,7 +203,8 @@ func onTruncateTable(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		job.State = model.JobStateCancelled
 		return ver, errors.Trace(err)
 	}
-	tblInfo, err := getTableInfo(t, job, schemaID)
```
Comment (winoros) [inline ddl/table.go]: Why change here?

Comment (lysu) [reply ddl/table.go]: before rebase ...`go tool vet` will say this shadow `err`...but after rebase complain is disappeared, orz....

reverted..

**Comment by lysu** [comment]
PTAL @XuHuaiyu

**Pull request merged**